In [246]:
import numpy as np
import pandas as pd
import scipy.stats
import statistics
import matplotlib.pyplot as plt
import seaborn as sns
%matplotlib inline

In [248]:
data_one = pd.read_csv('data_to_clean.csv')

In [249]:
## scrub journal title of spaces and caps
data_group = data_one.groupby('Journal title').count()
data_group.nlargest(5, 'Article title')
data_one['Journal title']=data_one['Journal title'].str.lower()
data_one['Journal title'] = data_one['Journal title'].str.strip()
data_group_2 = data_one.groupby('Journal title').count()
data_group_2.nlargest(5, 'Article title')

PMID/PMCID  Publisher  Article title  \
Journal title                                                           
plos one                                188        190            190   
journal of biological chemistry          52         53             53   
neuroimage                               28         29             29   
nucleic acids research                   25         26             26   
plos genetics                            23         24             24   

                                 COST (£) charged to Wellcome (inc VAT when charged)  
Journal title                                                                         
plos one                                                                       190    
journal of biological chemistry                                                 53    
neuroimage                                                                      29    
nucleic acids research                                                          26    
plos genetics                                                                   24

Atempting to group the Jornal Titles reveal they need to be srubbed for capitlizations. Will also srub spaces to ensure we get ther right results

The five Journals with the most articles are
- Plos One with 190 Articles
- Journal of Biological Chemistry with 53 articles
- Nueroimage with 29 articles
- Plos Genetics with 26 articles
- Plos Pathogens with 24 articles

In [250]:
### removing dollar and british pound signs and convert to float
### how to winzor
data_one['COST (£) charged to Wellcome (inc VAT when charged)'] = data_one['COST (£) charged to Wellcome (inc VAT when charged)'].str.replace('$', '')
data_one['COST (£) charged to Wellcome (inc VAT when charged)'] = data_one['COST (£) charged to Wellcome (inc VAT when charged)'].str.replace('\xA3', '')
data_one['COST (£) charged to Wellcome (inc VAT when charged)'] = data_one['COST (£) charged to Wellcome (inc VAT when charged)'].astype(float)

In [261]:
data_by_journal=data_one.groupby('Journal title')['COST (£) charged to Wellcome (inc VAT when charged)'].mean()
print('Average Cost of Plos One:','\xA3',str(round(data_by_journal.loc['plos one'],2)))
print('Average Cost of Journal of Biological Chemistry:','\xA3',str(round(data_by_journal.loc['journal of biological chemistry'],2)))
print('Average Cost of Neuroimage:','\xA3',str(round(data_by_journal.loc['neuroimage'],2)))
print('Average Cost of Nucleic Acids Research:','\xA3',str(round(data_by_journal.loc['nucleic acids research'],2)))
print('Average Cost of Plos Genetics:','\xA3',str(round(data_by_journal.loc['plos genetics'],2)))

print('Average Cost of all journals is ',data_one.loc[:,'COST (£) charged to Wellcome (inc VAT when charged)'].mean())

Average Cost of Plos One: £ 38751.29
Average Cost of Journal of Biological Chemistry: £ 20264.63
Average Cost of Neuroimage: £ 2215.17
Average Cost of Nucleic Acids Research: £ 1149.0
Average Cost of Plos Genetics: £ 84839.44
Average Cost of all journals is  24067.339971791254


In [262]:
data_by_journal=data_one.groupby('Journal title')['COST (£) charged to Wellcome (inc VAT when charged)'].median()
print('Median Cost of Plos One:','\xA3',str(round(data_by_journal.loc['plos one'],2)))
print('Median Cost of Journal of Biological Chemistry:','\xA3',str(round(data_by_journal.loc['journal of biological chemistry'],2)))
print('Median Cost of Neuroimage:','\xA3',str(round(data_by_journal.loc['neuroimage'],2)))
print('Median Cost of Nucleic Acids Research:','\xA3',str(round(data_by_journal.loc['nucleic acids research'],2)))
print('Median Cost of Plos Genetics:','\xA3',str(round(data_by_journal.loc['plos genetics'],2)))


print('Median cost of all journals is ',data_one.loc[:,'COST (£) charged to Wellcome (inc VAT when charged)'].median())

Median Cost of Plos One: £ 898.48
Median Cost of Journal of Biological Chemistry: £ 1314.53
Median Cost of Neuroimage: £ 2326.43
Median Cost of Nucleic Acids Research: £ 852.0
Median Cost of Plos Genetics: £ 1718.39
Median cost of all journals is  1884.01


In [259]:
print('The STD of all jdata_one.loc[:,'COST (£) charged to Wellcome (inc VAT when charged)'].std()

146860.66555904722